pkg/cmd/server/origin/master.go

```diff
@@ -347,35 +348,67 @@ func (c *MasterConfig) InstallProtectedAPI(container *restful.Container) ([]stri
 	initReadinessCheckRoute(root, "/healthz/ready", c.ProjectAuthorizationCache.ReadyForAccess)
 	initVersionRoute(container, "/version/openshift")
 
+	// Set up OAuth metadata only if we are configured to use OAuth
+	if c.Options.OAuthConfig != nil {
+		initOAuthAuthorizationServerMetadataRoute(container, "/.well-known/oauth-authorization-server", c.Options.OAuthConfig.MasterPublicURL)
```

Comment:
Make this a package private constant and add some nice godoc referencing the rfc